### PR TITLE
Added Kafka readers closing using WaitGroup

### PR DESCRIPTION
**Pull request by Josca**
- also added testing upload loop

### CODE REVIEW

File: base/mqueue/mqueue.go [hunk 1]
```diff
@@ -91,6 +92,9 @@ func (t *readerImpl) HandleMessages(handler MessageHandler) {
 	for {
 		m, err := t.FetchMessage(base.Context)
 		if err != nil {
+			if err.Error() == "context canceled" {
```
Comment (semtexzv) [inline base/mqueue/mqueue.go:95]: Please extract the string to a constant

Comment (Josca) [reply base/mqueue/mqueue.go]: done

File: evaluator/evaluate.go [hunk 2]
```diff
@@ -452,11 +453,14 @@ func run(readerBuilder mqueue.CreateReader) {
 	// We create multiple consumers, and hope that the partition rebalancing
 	// algorithm assigns each consumer a single partition
 	for i := 0; i < consumerCount; i++ {
-		go mqueue.RunReader(evalTopic, readerBuilder, handler)
+		wg.Add(1)
```
Comment (semtexzv) [inline evaluator/evaluate.go:456]: Why increment here ? The RunReader should manage its contribution to the waitgroup.

Comment (Josca) [reply evaluator/evaluate.go:456]: If we put it there (inside go routine function) theoretically wg.Wait() calling in main thread could be reached earlier than first wg.Add(1) is called.

**Pull request merged**